### PR TITLE
Copy improvements in the SQL Database verified source

### LINKED ISSUE #720: make clear in docs that we support any sql database as a source
### Feature description

Our docs:
https://dlthub.com/docs/dlt-ecosystem/verified-sources/sql_database
are not clear that we support all sql alchemy dialects and does not give any examples

### Are you a dlt user?

Yes, I'm already a dlt user.

### Use case

ingesting data from production databases

### Proposed solution

Let's fix it as follows:
1. change title to +50 SQL Databases (or similar)
2. before we get into user guide add a section on supported databases. mention that we support any sql alchemy dialect.
3. take 15 most prominent engines from https://docs.sqlalchemy.org/en/20/dialects/ and list them, then link to this list
4. mention that there are many unofficial dialects and link to duckdb-engine as an example
5. provide more info on how to create a connection string for a given dialect. (just extend existing docs)

### Related issues

_No response_

### PR DESCRIPTION
**Pull request by anuunchin**
### Description
Included the info asked in the issue.
Made some additional wording improvements in the docs on the SQL Database verified source.
5th bullet point in the issue, namely, providing more info on how to create a connection string for a given dialect, is still to be resolved. 

### Related Issues

- Partly resolves #720.

### Additional Context

Will need more clarification regarding the 5th point.

### CODE REVIEW

**Review by rudolfix — COMMENTED** [review]
LGTM! thanks for fixing spelling errors, I'll let @burnash to do a final review before we merge

File: docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md
Comment (burnash) [inline docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md:1]:
```suggestion
# 30+ SQL Databases
```

File: docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md
Comment (burnash) [inline docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md:14]: I suggest using a tip [admonition](https://docusaurus.io/docs/markdown-features/admonitions) instead of the markdown blockquote.

File: docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md
Comment (burnash) [inline docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md:44]: Same here: it's a good place to use a note [admonition](https://docusaurus.io/docs/markdown-features/admonitions) instead of the markdown blockquote.

File: docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md
Comment (burnash) [inline docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md:54]:
```suggestion
The database above doesn't require a password.
```

File: docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md
Comment (burnash) [inline docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md:122]: That's also a good place to use a tip admonition.

File: docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md
Comment (burnash) [inline docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md:188]: It's best to either remove a blockquote or use a "note" admonition here.

File: docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md
Comment (burnash) [inline docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md:345]:
```suggestion
   In this example, we load data from the `family` table, using the `updated` column for incremental loading. In the first run, the process loads all data starting from midnight (00:00:00) on January 1, 2022. Subsequent runs perform incremental loading, guided by the values in the `updated` field.
```

File: docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md
Comment (burnash) [inline docs/website/docs/dlt-ecosystem/verified-sources/sql_database.md:361]:
```suggestion
   This process initially loads all data from the `family` table starting at midnight on January 1, 2022. For later runs, it uses the `updated` field for incremental loading as well. 
```

**Review by burnash — CHANGES_REQUESTED** [review]
Hey @anuunchin thanks for great work! I've gone through your changes and overall, it's shaping up really well. There are a few minor adjustments I've suggested in the comments which I believe could make this page even better. Keep up the great work!

**Review by burnash — APPROVED** [review]
Thank you @anuunchin for the contribution. I appreciate the attention to details. Great work!

**Pull request merged**
